app/Providers/AuthServiceProvider.php

```diff
@@ -56,5 +56,21 @@ public function boot()
             }
             return false;
         });
+
+        Gate::define('create-product', function ($user) {
+            return $user->user_type === 'ADMIN_COOP' && $user->cooperative;
+        });
+
+        Gate::define('manage-product', function ($user, \App\Product $resource) {
+           return $user->user_type === 'ADMIN_COOP'
```

Comment:
Esse ponto aqui ficou redundante. Inicialmente adicionamos vários papéis como atributo tendo em vista realizar a autorização. Logo mais, no decorrer da implementação de alguns esquemas de autorização, percebi, junto a uma explicação que relembrei sobre a abstração do papel a partir do banco de dados dada por Gustavo a nós em nossa primeira (ou seria segunda?) reunião, que era possível inferir quem é o cooperativista checando se o usuário tem uma cooperativa associada ($user->cooperative ou $user->cooperative()->get()), se o usuário tem uma cooperativa associada ele é um cooperativista. Você pode fazer desse jeito, ou desse jeito ($user->user_type === 'ADMIN_COOP'), mas usar as duas formas fica redundante.

Reply:
Refatorei as policies